### PR TITLE
usm: sharedlibraries: Handle open(2) syscall

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

We currently only handle shared libraries being opened with openat(2) and openat2(2), which works fine with glibc, but musl (default on Alpine Linux, for example) uses open(2) so handle that too.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-989

### Describe how to test/QA your changes

Test cases are included in the PR.

### CODE REVIEW

File: pkg/network/usm/monitor_tls_test.go [hunk 1]
```diff
@@ -94,24 +95,30 @@ func (s *tlsSuite) TestHTTPSViaLibraryIntegration() {
...
-
-			for _, lib := range tlsLibs {
-				libSSLPath := lib.FindString(string(linked))
-				if _, err := os.Stat(libSSLPath); err == nil {
-					tests[index].prefetchLibs = append(tests[index].prefetchLibs, libSSLPath)
-				}
-			}
-		}
+		{
+			name:     "curl (musl)",
```
Comment (guyarb) [inline pkg/network/usm/monitor_tls_test.go:99]: let's have an inline documentation

**Review by guyarb — APPROVED** [review]

File: pkg/network/usm/sharedlibraries/ebpf.go [hunk 2]
```diff
@@ -177,12 +179,17 @@ func sysOpenAt2Supported() bool {
...
-// getSysOpenHooksIdentifiers returns the enter and exit tracepoints for openat and openat2 (if supported).
+// getSysOpenHooksIdentifiers returns the enter and exit tracepoints for supported open*
+// system calls.
 func getSysOpenHooksIdentifiers() []manager.ProbeIdentificationPair {
 	openatProbes := []string{openatSysCall}
 	if sysOpenAt2Supported() {
 		openatProbes = append(openatProbes, openat2SysCall)
 	}
+	// arm64 doesn't have open(2)
+	if runtime.GOARCH == "amd64" {
```
Comment (nplanel) [inline pkg/network/usm/sharedlibraries/ebpf.go:190]: may mark it explicitly like the commend
`if runtime.GOARCH != "arm64" {`

File: pkg/network/usm/monitor_tls_test.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/network/usm/monitor_tls_test.go]: Added.

**Pull request merged**